mgmt/ProcessManager.cc

```diff
@@ -126,7 +126,9 @@ ProcessManager::stop()
     ats_free(sig);
   }
 
-  ats_free(mgmt_signal_queue);
+  LLQ *tmp_queue    = mgmt_signal_queue;
+  mgmt_signal_queue = nullptr;
+  delete_queue(tmp_queue);
```

Comment:
I agree with `delete_queue()` should be used here. But why we can't delete mgmt_signal_queue directly?

Reply:
Never mind. Looked around `ProcessManager::stop()`, there're code like this.

Reply:
I think it's to avoid race conditions - setting `mgmt_signal_queue` to `nullptr` is effectively atomic, but deleting the queue is not.

Reply:
Make sense. Thanks for lecture!